### PR TITLE
Honda: 2024 CR-V Hybrid Car Port

### PR DESCRIPTION
**Pull request by vanillagorillaa**
**Checklist**

- [ ] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [X] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [X] route with openpilot: 197afc441c1f8c4f/2024-02-21--11-18-08
- [X] route with stock system: 197afc441c1f8c4f/2024-02-20--21-46-35
- [X] car harness used (if comma doesn't sell it, put N/A): Bosch C

TODO: Fix regen paddle issue (https://github.com/commaai/opendbc/issues/1100)

### CODE REVIEW

File: selfdrive/car/honda/fingerprints.py [hunk 1]
```diff
@@ -1433,4 +1433,9 @@
       b'37805-64S-AA10\x00\x00',
     ],
   },
+  CAR.CRV_HYBRID_6G: {
+    (Ecu.fwdRadar, 0x18dab0f1, None): [
+      b'8S302-3D4-A050\x00\x00',
+    ],
+  },
```
Comment (sshane) [inline selfdrive/car/honda/fingerprints.py:1440]: Any idea why there's only one ECU? Can we do an ECU scan on this car to search for other addresses?

File: selfdrive/car/honda/carstate.py [hunk 2]
```diff
@@ -267,7 +267,7 @@ def update(self, cp, cp_cam, cp_body):
 
   def get_can_parser(self, CP):
     messages = get_can_messages(CP, self.gearbox_msg)
-    return CANParser(DBC[CP.carFingerprint]["pt"], messages, get_pt_bus(CP.carFingerprint))
+    return CANParser(DBC[CP.carFingerprint]["pt"], messages, CanBus(CP).pt)
```
Comment (sshane) [inline selfdrive/car/honda/carstate.py:270]: Can you open a new PR that adds this CanBus class for Bosch/Nidec?

File: selfdrive/car/honda/fingerprints.py (hunk 1 above)
Comment (vanillagorillaa) [reply selfdrive/car/honda/fingerprints.py:1440]: the first route I got only had this version. I have newer routes that have the fwdcamera, but nothing else

File: selfdrive/car/honda/carstate.py (hunk 2 above)
Comment (vanillagorillaa) [reply selfdrive/car/honda/carstate.py:270]: sure, can do

**Comment by solaris7** [comment]
I own a 2024 CRV hybrid and I am a software engineer. I'd be happy to install this and troubleshoot this. What are some outstanding things that need to be done to make this work?

**Comment by vanillagorillaa** [comment]
> I own a 2024 CRV hybrid and I am a software engineer. I'd be happy to install this and troubleshoot this. What are some outstanding things that need to be done to make this work?

TODO: Fix regen paddle issue (https://github.com/commaai/openpilot/issues/30559)

Been waiting to finish this car (and the Pilot) once the Bosch C harness is for sale on the shop and we can get someone with the new accord set up (these three cars are Honda's new CANFD platform).

**Comment by sshane** [comment]
We've moved the car interfacing code to our [`opendbc`](https://github.com/commaai/opendbc) repository, which is now the new home for car ports and fingerprints. Please re-open your pull request against [`opendbc`](https://github.com/commaai/opendbc) at your convenience by using this command below. This will transform all changes under `selfdrive/car/` to `opendbc_repo/opendbc/car/`. Make sure you have initialized submodules beforehand and have checked out the latest `opendbc` commit.

```bash
PR_NUMBER=33045
curl -L https://github.com/commaai/openpilot/pull/$PR_NUMBER.patch | sed -e 's/selfdrive\/car/opendbc_repo\/opendbc\/car/g' | git apply -v --reject
```

Simply replace the PR number with your own. Once done, add the files, fix any conflicts, and open a new PR. Alternatively, you may start a new PR from scratch if that is easier for you.

**Pull request closed without merging**